airflow/providers/smtp/notifications/smtp.py

```diff
@@ -47,31 +50,41 @@ class SmtpNotifier(BaseNotifier):
 
     def __init__(
         self,
-        from_email: str | None,
+        *,
```

Comment:
This introduces a potential breaking change for any uses of SmtpNotifier that rely on positional arguments. Thoughts? Alternatively, we could just keep it as is and require that users pass it like was done before.

Reply:
Yeah. let's treat it as breaking change (however this one is most cases will be either super-easy to fix or will not break as people will already use keyword parameters.

Reply:
I don't see any need for this change; we can add only the newly added parameters after the `*` with a TODO comment for this change in the next major release.

Reply:
Ack @hussein-awala - I can revert. The one newly added parameter is `template` so that one parameter will come after the `*`. And will get the TODO comment in there for the next major release.

Reply:
This unfortunately means users will have to continue providing the `from_email` argument until then but that is not too much of a hassle IMO.